front/src/modules/pathfinding/utils.ts

```diff
@@ -34,21 +34,20 @@ export const formatSuggestedOperationalPoints = (
     coordinates: getPointCoordinates(geometry, pathLength, op.position),
   }));
 
-export const matchPathStepAndOp = (step: PathStep, op: SuggestedOP) => {
+export const matchPathStepAndOp = (
+  step: PathStep,
+  op: Pick<SuggestedOP, 'opId' | 'uic' | 'ch' | 'trigram' | 'track' | 'offsetOnTrack'>
+) => {
   if ('operational_point' in step) {
     return step.operational_point === op.opId;
```

Comment:
You added a todo comment for the other 2 functions when matching based on id, should it be added to matchPathStepAndOp too?

Reply:
There are two different `opId`-based checks:

- `pathStep.id === op.opId`: here we abuse `PathItemLocation.id` (which identifies a waypoint, not an OP). Path item location IDs and OP IDs are two different namespaces.
- `step.operational_point === op.opId`: a `PathItemLocation` variant which describes the waypoint with an OP ID, instead of a UIC+ch, or trigram+ch, etc.

So a `PathItemLocation` could look like this:

```ts
{
    id: 'waypoint42',
    operational_point: 'op69'
}
```

Reply:
Sure, thanks!